pyrsmi/rocml.py

```diff
@@ -544,26 +593,172 @@ def _init_processor_handles():
     
     _processor_handles = []
     
+    # Try the native amdsmi Python package first (best for containers)
+    if _init_processor_handles_via_amdsmi_package():
+        _handle_initialized = True
+        logging.info(f'Initialized {len(_processor_handles)} GPU processor handles (via amdsmi package)')
+        return
+    
+    # Try the newer amdsmi_get_processor_handles_by_type API
+    if _init_processor_handles_by_type():
+        _handle_initialized = True
+        logging.info(f'Initialized {len(_processor_handles)} GPU processor handles (by type)')
+        return
+    
+    # Fallback to socket-based enumeration
+    logging.debug('Falling back to socket-based processor enumeration')
+    if _init_processor_handles_by_socket():
+        _handle_initialized = True
+        logging.info(f'Initialized {len(_processor_handles)} GPU processor handles (by socket)')
+        return
+    
+    logging.warning('No AMD GPU processors found')
+
+
+def _init_processor_handles_via_amdsmi_package():
+    """Initialize processor handles using the native amdsmi Python package.
+    
+    This is the preferred method for containerized environments where the
+    amdsmi Python package is installed. It's a first-party AMD package that
+    handles GPU discovery correctly in containers.
+    
+    @return: True if successful and found at least one GPU, False otherwise
+    """
+    global _processor_handles
+    
+    try:
+        # Try to import the native amdsmi package
+        import amdsmi
+        
+        # Initialize amdsmi
+        try:
+            amdsmi.amdsmi_init()
+        except Exception as e:
```

Comment:
Can we say anything more about what kinds of exceptions are expected? I we don't know at all, maybe logging.error or logging.warning at least? (similarly below)